### PR TITLE
Boost: Improve active module list in speed score requests

### PR DESCRIPTION
**Pull request by haqadn**
Speed score requests store which modules are active at the time of the request. This is to be able to compare it with the score of `no-boost` modules active and figuring out the impact of boost modules. Previously, it was just looking at what modules are enabled. But, this may not always be enough.

Example:
- Critical CSS and Cloud CSS need to generate the CSS before it can be useful.
- Cache engine needs to load successfully for it to help optimize the site.

## Proposed changes:

* In Speed Score requests, keep record of the modules that are actually working, just being enabled is not enough.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
None

## Does this pull request change what data or activity we track or use?
None

## Testing instructions:
* Enable cache.
* Rig the `advanced-cache.php` file so the caching engine doesn't load.
* Run a speed score and check if `page_cache` is stored as an `active_module` in [`Speed_Score_Request`](https://github.com/Automattic/jetpack/blob/9f7b71faae30adb636c47d29aacd8be4fc44b679/projects/packages/boost-speed-score/src/class-speed-score-request.php#L17).

### CODE REVIEW

File: projects/packages/boost-speed-score/src/class-speed-score-request.php [hunk 1]
```diff
@@ -105,21 +105,10 @@ public static function generate_cache_id_from_url( $url ) {
...
-			'critical_css',
-			'image_cdn',
-			'minify_css',
-			'minify_js',
-			'render_blocking_js',
-		);
-
-		return array_intersect( $this->active_modules, $performance_modules );
+		return $this->active_modules;
 	}
```
Comment (thingalon) [inline projects/packages/boost-speed-score/src/class-speed-score-request.php]: This function used to differentiate between performance modules and non-performance modules (e.g.: image guide). `performance_modules` is still specified in the name.

Should this function be renamed?

File: projects/packages/jetpack-mu-wpcom/.phan/baseline.php [hunk 2]
```diff
@@ -9,29 +9,31 @@
  */
 return [
     // # Issue statistics:
-    // PhanUndeclaredFunction : 75+ occurrences
+    // PhanUndeclaredFunction : 100+ occurrences
```
Comment (thingalon) [inline projects/packages/jetpack-mu-wpcom/.phan/baseline.php:12]: What is phan baseline? Is this stuff we should be changing? It's in another package? :-|

File: projects/plugins/boost/app/modules/Modules_Setup.php [hunk 3]
```diff
@@ -38,6 +38,21 @@ public function have_enabled_modules() {
...
+	 */
+	public function get_active_optimization_modules() {
+		$working_modules = array();
+		foreach ( $this->available_modules as $slug => $module ) {
+			if ( $module->is_optimizing() ) {
+				$working_modules[] = $slug;
+			}
+		}
+		return $working_modules;
+	}
```
Comment (thingalon) [inline projects/plugins/boost/app/modules/Modules_Setup.php]: `get_active_optimization_modules` returns the ones which are `optimizing`. So ... 

If Critical CSS is enabled, but has not generated Critical CSS successfully, it is active and it is an optimization module but this function won't return it. 

`get_optimizing_modules()` perhaps? Even that's a bit complex.

I'm not sure what to call this, but I think the current name is a bit misleading.

File: projects/plugins/boost/app/modules/optimizations/cloud-css/Cloud_CSS.php [hunk 4]
```diff
@@ -51,6 +52,15 @@ public function setup() {
...
 	}
 
+	/**
+	 * Check if the module is ready and already serving critical CSS.
+	 *
+	 * @return bool
+	 */
+	public function is_ready() {
+		return ( new Critical_CSS_State() )->is_generated();
+	}
```
Comment (thingalon) [inline projects/plugins/boost/app/modules/optimizations/cloud-css/Cloud_CSS.php:62]: If this is called `ready` perhaps the above "get_active_optimizatoin_modules` should be `get_ready_optimizations` ? 

I feel like this PR is introducing `ready` as a concept, so the name should be used consistently.

**Review by thingalon — CHANGES_REQUESTED** [review]
Overall I like the approach of defining an interface for Optimizations which is then able to answer questions like "are you ready?"

I think there is some confusing nomenclature, which could lead to confusing code - easier to fix it now. :)

File: projects/packages/jetpack-mu-wpcom/.phan/baseline.php (hunk 2 above)
Comment (haqadn) [reply projects/packages/jetpack-mu-wpcom/.phan/baseline.php:12]: This is auto-generated by `phan`, which I am not having "fun" with atm. Yes, that was a "pun".

**phan** was added for static analysis in mono-repo recently.

**Review by thingalon — APPROVED** [review]
Ok thanks for cleaning up the nomenclature :)

**Pull request merged**